pandas/core/reshape/merge.py

```diff
@@ -1200,9 +1219,32 @@ def _maybe_coerce_merge_keys(self):
                 typ = rk.categories.dtype if rk_is_cat else object
                 self.right = self.right.assign(**{name: self.right[name].astype(typ)})
 
+    def _create_cross_configuration(
+        self, _left, _right
+    ) -> Tuple["DataFrame", "DataFrame", str, str]:
+        cross_col = f"_cross_{hashlib.md5().hexdigest()}"
+        how = "inner"
+        return (
+            _left.assign(**{cross_col: 1}),
+            _right.assign(**{cross_col: 1}),
+            how,
+            cross_col,
+        )
+
     def _validate_specification(self):
+        if self.how == "cross":
+            if (
+                self.left_index
+                or self.right_index
+                or self.right_on is not None
+                or self.left_on is not None
+                or self.on is not None
+            ):
+                raise MergeError(
+                    "Can not pass any merge columns when using cross as merge method"
```

Comment:
can you say that left_on,right_on,on must be None, and left_index,right_index must be False

Reply:
Thx, done